consensus/moca/proposal.go

```diff
@@ -298,6 +298,11 @@ func (consensus *Consensus) requestProposal(neighbor *node.RemoteNode, blockHash
 		return nil, err
 	}
 
+	receivedBlockHash := block.Header.Hash()
+	if receivedBlockHash != blockHash {
```

Comment:
Why pre-verified block hash at here? There are block verification in moca voting and persistblock already.

Reply:
This verification ensures that only requested block can be sent as response (as the request proposal logic) and could potentially prevent some attack in the future.

Reply:
Anyway, just a little expenses for hash compare. It is acceptable for avoid requestProposal spoofing.